test/commands.js

```diff
@@ -437,6 +439,78 @@ let runTransferFromCommand = async (command, state) => {
   return state;
 }
 
+
+//
+// Market Maker commands
+//
+
+let priceFactor = 100000
+
+let getMMMaxClaimableEth = function(state) {
+  if (state.marketMakerMonth >= state.marketMakerPeriods) {
+    help.debug("calculating maxClaimableEth with", state.marketMakerStartingBalance,
+      state.marketMakerClaimedEth,
+      state.returnedWeiForBurnedTokens);
+    return state.marketMakerStartingBalance.
+      minus(state.marketMakerClaimedEth).
+      minus(state.returnedWeiForBurnedTokens);
+  } else {
+    const maxClaimable = state.marketMakerStartingBalance.
+      mul(state.claimablePercentage).dividedBy(priceFactor).
+      mul(state.initialTokenSupply - state.marketMakerBurnedTokens).
+      dividedBy(state.initialTokenSupply).
+      minus(state.marketMakerClaimedEth);
+    return _.max([0, maxClaimable]);
+  }
+}
+
+// TODO: implement finished
+let isMarketMakerFinished = (state) => false
+
+let runMarketMakerSendTokensCommand = async (command, state) => {
+  if (state.marketMaker === undefined) {
+    // doesn't make sense to execute the actual command, let's just assert
+    // that the crowdsale was not funded (in which case there should be MM)
+    assert.equal(false, state.crowdsaleFinalized && state.crowdsaleFunded,
+      "if there's no market Maker, crowdsale should not have been funded");
```

Comment:
we can reach the minCap of 5M usd but not the 10M usd needed for the MM creation

Reply:
good catch, I'll fix it

Reply:
by the way, one problem we are having with adding property based tests for the market maker is that the sequence of commands usually make it so the market maker is not created, even the crowdsale is not finished and funded, so in practical terms this is the branch that's executed 99.9999% of the times. It'd be nice to find a way to have the crowdsale to finish funded more frequently in the tests

Reply:
Expanded the comment in 23bc842, not the full solution but good enough for now?